src/MLIR/AST.hs

```diff
@@ -104,10 +104,10 @@ instance Eq Type where
 data Location =
     UnknownLocation
   | FileLocation { locPath :: BS.ByteString, locLine :: UInt, locColumn :: UInt }
+  | NameLocation { name :: BS.ByteString, childLoc :: Location }
```

Comment:
Unfortunately names of record fields become top-level functions in Haskell which is why it's fairly common to come up with prefixes that are uniform for a single `data` declaration. I've been using `loc` as the prefix here (`locPath`, `locLine`, ...). Can we please use the same convention (`locName`, `locChild`)?

Reply:
Good point, updated, thanks